src/deroff.rs

```diff
@@ -296,6 +326,152 @@ impl Deroffer {
             c => c.chars().all(|c| c.is_whitespace()),
         }
     }
+
+    // Replaces the g_macro_dict lookup in the Python code
+    fn g_macro_dispatch<'a>(&mut self, s: &'a str) -> bool {
+        match s {
+            "SH" => self.macro_sh(),
+            "SS" => self.macro_ss_ip(),
+            "IP" => self.macro_ss_ip(),
+            "H " => self.macro_ss_ip(),
+            "I " => self.macro_i_ir(),
+            "IR" => self.macro_i_ir(),
+            "IB" => self.macro_i_ir(),
+            "B " => self.macro_i_ir(),
+            "BR" => self.macro_i_ir(),
+            "BI" => self.macro_i_ir(),
+            "R " => self.macro_i_ir(),
+            "RB" => self.macro_i_ir(),
+            "RI" => self.macro_i_ir(),
+            "AB" => self.macro_i_ir(),
+            "Nm" => self.macro_nm(),
+            "] " => self.macro_close_bracket(),
+            "PS" => self.macro_ps(),
+            "PE" => self.macro_pe(),
+            "TS" => self.macro_ts(),
+            "T&" => self.macro_t_and(),
+            "TE" => self.macro_te(),
+            "EQ" => self.macro_eq(),
+            "EN" => self.macro_en(),
+            "R1" => self.macro_r1(),
+            "R2" => self.macro_r2(),
+            "de" => self.macro_de(),
+            "BL" => self.macro_bl_vl(),
+            "VL" => self.macro_bl_vl(),
+            "AL" => self.macro_bl_vl(),
+            "LB" => self.macro_bl_vl(),
+            "RL" => self.macro_bl_vl(),
+            "ML" => self.macro_bl_vl(),
+            "DL" => self.macro_bl_vl(),
+            "BV" => self.macro_bv(),
+            "LE" => self.macro_le(),
+            "LP" => self.macro_lp_pp(),
+            "PP" => self.macro_lp_pp(),
+            "P\n" => self.macro_lp_pp(),
+            "ds" => self.macro_ds(),
+            "so" => self.macro_so_nx(),
+            "nx" => self.macro_so_nx(),
+            "tr" => self.macro_tr(),
+            "sp" => self.macro_s(),
+            _ => self.macro_other(),
+        }
+    }
+
+    fn macro_sh(&mut self) -> bool {
+        false
```

Comment:
Might be more accurate for all of the default function bodies to be `unimplemented!()` rather than `false`.